plugins/database/mysql/connection_producer.go

```diff
@@ -13,14 +13,30 @@ import (
 	"sync"
 	"time"
 
+	cloudmysql "cloud.google.com/go/cloudsqlconn/mysql/mysql"
+
 	"github.com/go-sql-driver/mysql"
 	"github.com/hashicorp/go-secure-stdlib/parseutil"
 	"github.com/hashicorp/go-uuid"
+	"github.com/mitchellh/mapstructure"
+
 	"github.com/hashicorp/vault/sdk/database/helper/connutil"
 	"github.com/hashicorp/vault/sdk/database/helper/dbutil"
-	"github.com/mitchellh/mapstructure"
 )
 
+const (
+	authTypeIAM   = "iam"
+	cloudSQLMySQL = "cloudsql-mysql"
+	driverMySQL   = "mysql"
+)
+
+// ensure cloud driver registration only happens once.
+// the complication here is that registration happens at point X, and is not allowed to happen again, ever.
+// This cannot be stored as state within a connection producer, because those producers are config-specific,
+// so if we configure, say, two databases that are both cloud-mysql, we must only register once.
+// we might be able to cleverly do this with init().
+var onceler sync.Once
```

Comment:
might've missed it, but where is this being used?

Reply:
obsolete variable, good catch